src/Modules/UngradedModules/Presentation.jl

```diff
@@ -582,6 +589,75 @@ function prune_with_map(M::ModuleFP{T}) where {T<:Union{MPolyRingElem, MPolyQuoR
   return M_new, phi
 end
 
+function ensure_presentation_ctx!(M::SubquoModule)
+    ensure_presentation_ctx!(base_ring(M), M)
+end
+
+function ensure_presentation_ctx!(M::SubquoModule{ZZRingElem})
```

Comment:
as above, these `ensure_*` functions could be replaced by a getter function that then makes use of `@attr` for caching